### PR TITLE
Es6 lib fetch

### PR DESCRIPTION
**Pull request by nicl**
## What does this change?

More ES6ifying.

### CODE REVIEW

File: static/src/javascripts/lib/fetch.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import fetch from 'lib/fetch';
+import sinon from 'sinon';
+
+let xhr;
+let requests;
+
+describe('Fetch util', () => {
+    beforeEach(() => {
+        requests = [];
+        xhr = sinon.useFakeXMLHttpRequest();
```
Comment (nicl) [inline static/src/javascripts/lib/fetch.spec.js]: From what we understand, sinon clobbers the global xml request object and then we restore it to the original value in the afterEach block. @SiAdcock has suggested we might not even have to do this anymore as Jest sandboxes each test suite but I've kept it in for now.

File: static/src/javascripts/lib/fetch.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * - - headers are not populated
+ *
+ * If you're still wondering what it actually supports
+ * - CORS
+ * - credentials
+ * - body, any body that you want to add to your request except for GET or HEAD
+ * - response.text() and response.json()
+ * - response.ok .status .statusText
+ */
+const buildRequest = (path, options) => {
```
Comment (gustavpursche) [inline static/src/javascripts/lib/fetch.js]: Could you add the flow-types here and in all other functions please? :)

File: package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
     "raven-js": "^3.14.0",
     "react": "0.13.2",
     "reqwest": "2.0.5",
+    "sinon": "^2.1.0",
```
Comment (sndrs) [inline package.json]: should be a devdep?

Comment (nicl) [inline package.json]: 👍

Comment (nicl) [inline package.json]: At least, it's just for tests yes.

File: static/src/javascripts/lib/fetch.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+const fetch = (input: string, init: ?Object) =>
+    new Promise((resolve, reject) => {
+        const req = buildRequest(input, init || {});
+        reqwest(req)
+            .then(resp => {
+                resolve(createResponse(resp));
+            })
+            .fail(resp => {
```
Comment (gustavpursche) [inline static/src/javascripts/lib/fetch.js]: `.fail` -> `.catch` to be more Promise A+ compliant.

File: static/src/javascripts/lib/fetch.js (hunk 2 above)
Comment (nicl) [inline static/src/javascripts/lib/fetch.js]: Yeah sounds good. I guess we're currently only requiring it for exported functions?

Comment (gustavpursche) [inline static/src/javascripts/lib/fetch.js]: Yeah, that's the minimum we are trying to achieve. For `lib/*.js` we wanted to be a bit more ambitious :)

**Comment by nicl** [comment]
@gustavpursche thanks for the feedback. Does this look okay now?

**Review by SiAdcock — APPROVED** [review]
LGTM 👍

**Pull request merged**